services/121-service/src/migrate-visa/migrate-visa.service.ts

```diff
@@ -0,0 +1,689 @@
+      { name: 'Tenant-ID', value: process.env.INTERSOLVE_VISA_TENANT_ID },
+    ];
+    // On success this returns a 204 No Content
+    return await this.httpService.post<any>(url, payload, headers);
+  }
+
+  private isSuccessResponseStatus(status: number): boolean {
+    return status >= 200 && status < 300;
+  }
+
+  public async postToggleBlockWallet(
+    tokenCode: string | null,
+    payload: {
+      reasonCode: IntersolveBlockTokenReasonCodeEnum;
+    },
+    block: boolean,
+  ): Promise<{
+    data: {
+      success?: boolean;
+      errors?: ErrorsInResponseIntersolveApi[];
+      code?: string;
+      correlationId?: string;
+    };
+    status: number;
+    statusText?: string;
+  }> {
+    const authToken = await this.getAuthenticationToken();
+    const apiPath = process.env.INTERSOLVE_VISA_PROD
+      ? 'pointofsale-payments'
+      : 'pointofsale';
+    const url = `${intersolveVisaApiUrl}/${apiPath}/v1/tokens/${tokenCode}/${
+      block ? 'block' : 'unblock'
+    }`;
+    const headers = [
+      { name: 'Authorization', value: `Bearer ${authToken}` },
+      { name: 'Tenant-ID', value: process.env.INTERSOLVE_VISA_TENANT_ID },
+    ];
+    const blockResult = await this.httpService.post<any>(url, payload, headers);
+    const result = {
+      status: blockResult.status,
+      statusText: blockResult.statusText,
+      data: blockResult.data,
+    };
+    return result;
+  }
+
+  public async postLinkToken(
+    childTokenCode: string | null,
+    parentTokenCode: string | null,
+  ): Promise<{
+    data: {
+      success?: boolean;
+      errors?: ErrorsInResponseIntersolveApi[];
+      code?: string;
+      correlationId?: string;
+    };
+    status: number;
+    statusText?: string;
+  }> {
+    const authToken = await this.getAuthenticationToken();
+    const apiPath = process.env.INTERSOLVE_VISA_PROD
+      ? 'wallet-payments'
+      : 'wallet';
+    const url = `${intersolveVisaApiUrl}/${apiPath}/v1/tokens/${parentTokenCode}/link-token`;
+    const headers = [
+      { name: 'Authorization', value: `Bearer ${authToken}` },
+      { name: 'Tenant-ID', value: process.env.INTERSOLVE_VISA_TENANT_ID },
+    ];
+    const payload = {
+      tokenCode: childTokenCode,
+    };
+    const linkResult = await this.httpService.post<any>(url, payload, headers);
+    return linkResult;
+  }
+
+  public async postTransfer(
+    parentTokenCode: string,
+    amountInCent: number,
+  ): Promise<{
+    data: {
+      success?: boolean;
+      errors?: ErrorsInResponseIntersolveApi[];
+      code?: string;
+      correlationId?: string;
+    };
+    status: number;
+    statusText?: string;
+  }> {
+    const authToken = await this.getAuthenticationToken();
+    const apiPath = process.env.INTERSOLVE_VISA_PROD
+      ? 'wallet-payments'
+      : 'wallet';
+    const url = `${intersolveVisaApiUrl}/${apiPath}/v1/tokens/${process.env.INTERSOLVE_VISA_FUNDINGTOKEN_CODE}/transfer`;
```

Comment:
This should not be process.env but we should get it from the program fsp configurations @diderikvw